xla/backends/gpu/runtime/gpublas_lt_matmul_thunk.cc

```diff
@@ -35,117 +38,165 @@ limitations under the License.
-    d_scale = allocs.GetDeviceAddress(d_scale_buffer_);
+  if (d_scale_.allocation() != nullptr) {
+    d_scale = allocs.GetDeviceAddress(d_scale_);
   }
-  if (d_amax_buffer_.allocation() != nullptr) {
-    d_amax = allocs.GetDeviceAddress(d_amax_buffer_);
+  if (d_amax_.allocation() != nullptr) {
+    d_amax = allocs.GetDeviceAddress(d_amax_);
   }
-
-  se::DeviceMemoryBase aux;
-  if (aux_buffer_.allocation() != nullptr) {
-    aux = allocs.GetDeviceAddress(aux_buffer_);
+  if (aux_.allocation() != nullptr) {
+    aux = allocs.GetDeviceAddress(aux_);
   }
-
-  se::DeviceMemoryBase workspace;
-  if (workspace_buffer_.has_value()) {
-    workspace = allocs.GetDeviceAddress(workspace_buffer_.value());
+  if (workspace_.has_value()) {
+    workspace = allocs.GetDeviceAddress(workspace_.value());
   }
 
   return plan->ExecuteOnStream(
-      params.stream, allocs.GetDeviceAddress(a_buffer_),
-      allocs.GetDeviceAddress(b_buffer_), allocs.GetDeviceAddress(c_buffer_),
-      allocs.GetDeviceAddress(d_buffer_), bias, aux, a_scale, b_scale, c_scale,
-      d_scale, d_amax, algorithm, workspace);
+      stream, allocs.GetDeviceAddress(a_),
+      allocs.GetDeviceAddress(b_), allocs.GetDeviceAddress(c_),
+      allocs.GetDeviceAddress(d_), bias, aux, a_scale, b_scale, c_scale,
+      d_scale, d_amax, workspace);
 }
 
-absl::StatusOr<se::gpu::BlasLt::MatmulPlan*> CublasLtMatmulThunk::GetMatmulPlan(
-    const se::Stream* stream) {
-  {
-    absl::MutexLock lock(&matmul_plans_cache_mutex_);
-    auto it = matmul_plans_cache_.find(stream);
-    if (it != matmul_plans_cache_.end()) return it->second.get();
-  }
-  TF_ASSIGN_OR_RETURN(auto plan, se::gpu::BlasLt::GetMatmulPlan(
-                                     stream, gemm_config_, epilogue_));
-
-  absl::MutexLock lock(&matmul_plans_cache_mutex_);
-  auto [it, _] = matmul_plans_cache_.emplace(stream, std::move(plan));
-  return it->second.get();
-}
-
-absl::StatusOr<se::gpu::BlasLt::MatmulAlgorithm>
-CublasLtMatmulThunk::GetMatmulAlgorithm(const se::Stream* stream,
-                                        const se::gpu::BlasLt::MatmulPlan* plan,
-                                        int64_t max_workspace) {
-  {
-    absl::MutexLock lock(&matmul_algorithm_cache_mutex_);
-    auto it = matmul_algorithm_cache_.find(plan);
-    if (it != matmul_algorithm_cache_.end()) return it->second;
-  }
-  TF_ASSIGN_OR_RETURN(
-      auto algorithms,
-      plan->GetAlgorithms(stream,
-                          /*max_algorithm_count*/ 128,
-                          /*max_workspace_size*/ max_workspace));
-  TF_RET_CHECK(algorithm_idx_ >= 0 && algorithm_idx_ < algorithms.size());
-
-  absl::MutexLock lock(&matmul_algorithm_cache_mutex_);
-  auto [it, _] =
-      matmul_algorithm_cache_.emplace(plan, algorithms[algorithm_idx_]);
-  return it->second;
+auto CublasLtMatmulThunk::GetCachedMatmulPlan(
+    const ExecuteParams& params) -> absl::StatusOr<se::gpu::BlasLt::MatmulPlan *> {
```

Comment:
The Google style guide wants us to use the ordinary function syntax unless trailing return types are needed.